test/fit_tests/common/fit_common.py

```diff
@@ -0,0 +1,528 @@
+                                       )
+        if rest_action == "post":
+            result_data = requests.post(url_command,
+                                        data=json.dumps(rest_payload),
+                                        headers=rest_headers,
+                                        timeout=rest_timeout,
+                                        verify=sslverify
+                                        )
+        if rest_action == "binary-post":
+            result_data = requests.post(url_command,
+                                        data=rest_payload,
+                                        headers={"Content-Type": "application/octet-stream"},
+                                        timeout=rest_timeout,
+                                        verify=sslverify
+                                        )
+        if rest_action == "patch":
+            result_data = requests.patch(url_command,
+                                         data=json.dumps(rest_payload),
+                                         headers=rest_headers,
+                                         timeout=rest_timeout,
+                                         verify=sslverify
+                                         )
+    except requests.exceptions.Timeout:
+        return {'json':'', 'text':'',
+                'status':0,
+                'headers':'',
+                'timeout':True}
+
+    try:
+        result_data.json()
+    except ValueError:
+
+        if VERBOSITY >= 9:
+            print "restful: TEXT =\n"
+            print result_data.text
+        if VERBOSITY >= 4:
+            print "restful: Status code:", result_data.status_code, "\n"
+        return {'json':{}, 'text':result_data.text, 'status':result_data.status_code,
+                'headers':result_data.headers.get('content-type'),
+                'timeout':False}
+    else:
+
+        if VERBOSITY >= 9:
+            print "restful: JSON = \n"
+            print json.dumps(result_data.json(), sort_keys=True, indent=4)
+        if VERBOSITY >= 4:
+            print "restful: Status code =", result_data.status_code, "\n"
+        return {'json':result_data.json(), 'text':result_data.text,
+                'status':result_data.status_code,
+                'headers':result_data.headers.get('content-type'),
+                'timeout':False}
+
+
+# Get the list of BMC IP addresses that we can find
+def get_bmc_ips():
+    idlist = [] # list of unique dcmi node IDs
+    # If we have already done this, use that list
+    if len(BMC_LIST) == 0:
+        ipscan = remote_shell('arp')['stdout'].split()
+
+        for ipaddr in ipscan:
+            if ipaddr[0:3] == "172" and remote_shell('ping -c 1 -w 5 ' + ipaddr)['exitcode'] == 0:
+                # iterate through all known IPMI users
+                for item in GLOBAL_CONFIG['credentials']['bmc']:
+                    return_code = remote_shell('ipmitool -I lanplus -H ' + ipaddr + ' -U ' + item['username'] \
```

Comment:
I think wrap a util function to send IPMI command in remote shell will bring much benefit, as I see there are a lot other codes have similar code to send raw ipmi command.